src/stdatamodels/fits_support.py

```diff
@@ -509,6 +509,17 @@ def _save_extra_fits(hdulist, tree):
                     continue
                 hdu.header.append((key, val, comment), end=True)
 
+    def callback(node):
+        if isinstance(node, (np.ndarray, NDArrayType)):
+            # replace extra_fits arrays with special source values
+            # that represent links to that array in the FITS extension
+            for hdu_index, hdu in enumerate(hdulist):
+                if hdu.data is not None and node is hdu.data:
+                    return _create_tagged_dict_for_fits_array(hdu, hdu_index)
+        return node
+
+    return treeutil.walk_and_modify(tree, callback)
```

Comment:
Can we avoid this extra `walk_and_modify`? It deep copies the tree:
https://www.asdf-format.org/projects/asdf/en/latest/api/asdf.treeutil.walk_and_modify.html#asdf.treeutil.walk_and_modify

Reply:
What's the equivalent function for not deep copying and modifying in-place?  `iter_tree`?

Reply:
That's habit forming... Looks good to me!